System/Process.hs

```diff
@@ -703,15 +703,15 @@ waitForProcess ph@(ProcessHandle _ delegating_ctlc _) = lockWaitpid $ do
     OpenHandle h  -> do
         -- don't hold the MVar while we call c_waitForProcess...
         e <- waitForProcess' h
-        e' <- modifyProcessHandle ph $ \p_' ->
+        (e', was_open) <- modifyProcessHandle ph $ \p_' ->
```

Comment:
I'm honestly not following what the difference is in the previous and new version of the code, can you elaborate? It seems like the goal is to call `endDelegateControlC` outside of `modifyProcessHandle`, but I'm not sure why that makes a difference.

Reply:
The thing is that `endDelegateControlC` throws `UserInterrupt`, which means we're not setting the process handle to "closed". (`modifyProcessHandle` is just `modifyMvar`, which leaves the value as is if an exception is raised.)

Essentially, before was:

```
start modifyProcessHandle
  endDelegateControlC
    throw UserInterrupt
end modifyProcessHandle, actually writing handle (never reached!)
```

After is:

```
start modifyProcessHandle
end modifyProcessHandle, actually writing handle
endDelegateControlC
  throw UserInterrupt
```

Reply:
Got it. Could you add a comment to the `endDelegateControlC` call about the fact that it throws `UserInterupt`. I'd actually not noticed that personally (thus my question), and it would be a good thing to help avoid someone accidentally "improving" the code in the future.